### PR TITLE
updated ff docs to include and deprecate skip

### LINKED ISSUE #43865: Ansible hangs if remote host has a NFS issue.
##### SUMMARY
Ansible hangs up forever when remote host has a failed NFS volume.

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
setup module

##### ANSIBLE VERSION
```
ansible 2.5.5
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible-2.5.5-py2.7.egg/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, Jun 22 2018, 17:22:34) [GCC 4.4.7 20120313 (Red Hat 4.4.7-18)]
```

##### CONFIGURATION
```
[defaults]
gather_timeout = 10
host_key_checking = False
timeout = 10
action_plugins     = /usr/share/ansible/plugins/action
[inventory]
[privilege_escalation]
[paramiko_connection]
[ssh_connection]
[persistent_connection]
connect_timeout = 30
connect_retry_timeout = 15
command_timeout = 10
[accelerate]
[selinux]
[colors]
[diff]
```

##### OS / ENVIRONMENT
CentOS 6.9 x86_64

##### STEPS TO REPRODUCE
1. Mount a NFS volume to a remote machine;
2. Shutdown NFS service;
3. Run `ansible -m setup <REMOTE_MACHINE>`

##### EXPECTED RESULTS
The timeout works rather than hanging up there.

##### ACTUAL RESULTS
```
ansible 2.5.5
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/yingchun.huang/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible-2.5.5-py2.7.egg/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.15 (default, Jun 22 2018, 17:22:34) [GCC 4.4.7 20120313 (Red Hat 4.4.7-18)]
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
Parsed /tools/cmdb/short_inventory.txt inventory source with ini plugin
Loading callback plugin minimal of type stdout, v2.0 from /usr/local/lib/python2.7/site-packages/ansible-2.5.5-py2.7.egg/ansible/plugins/callback/minimal.pyc
META: ran handlers

Using module file /usr/local/lib/python2.7/site-packages/ansible-2.5.5-py2.7.egg/ansible/modules/system/setup.py
<cm075> ESTABLISH SSH CONNECTION FOR USER: root
<cm075> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/yingchun.huang/.ansible/cp/2091d1b79e cm075 '/bin/sh -c '"'"'echo ~root && sleep 0'"'"''
<cm075> (0, '/root\n', 'OpenSSH_5.3p1, OpenSSL 1.0.1e-fips 11 Feb 2013\ndebug1: Reading configuration data /home/yingchun.huang/.ssh/config\r\ndebug1: Applying options for *\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 4 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_request_forwards: requesting forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 8957\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 3\r\ndebug1: mux_client_request_session: master session id: 3\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<cm075> ESTABLISH SSH CONNECTION FOR USER: root
<cm075> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/yingchun.huang/.ansible/cp/2091d1b79e cm075 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1533799417.82-278701308886841 `" && echo ansible-tmp-1533799417.82-278701308886841="` echo /root/.ansible/tmp/ansible-tmp-1533799417.82-278701308886841 `" ) && sleep 0'"'"''
<cm075> (0, 'ansible-tmp-1533799417.82-278701308886841=/root/.ansible/tmp/ansible-tmp-1533799417.82-278701308886841\n', 'OpenSSH_5.3p1, OpenSSL 1.0.1e-fips 11 Feb 2013\ndebug1: Reading configuration data /home/yingchun.huang/.ssh/config\r\ndebug1: Applying options for *\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 4 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_request_forwards: requesting forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 8957\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 3\r\ndebug1: mux_client_request_session: master session id: 3\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<cm075> PUT /home/yingchun.huang/.ansible/tmp/ansible-local-89634L0CC2/tmpkgzcn9 TO /root/.ansible/tmp/ansible-tmp-1533799417.82-278701308886841/setup.py
<cm075> SSH: EXEC sftp -b - -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/yingchun.huang/.ansible/cp/2091d1b79e '[cm075]'
<cm075> (0, 'sftp> put /home/yingchun.huang/.ansible/tmp/ansible-local-89634L0CC2/tmpkgzcn9 /root/.ansible/tmp/ansible-tmp-1533799417.82-278701308886841/setup.py\nUploading /home/yingchun.huang/.ansible/tmp/ansible-local-89634L0CC2/tmpkgzcn9 to /root/.ansible/tmp/ansible-tmp-1533799417.82-278701308886841/setup.py\n', 'OpenSSH_5.3p1, OpenSSL 1.0.1e-fips 11 Feb 2013\ndebug1: Reading configuration data /home/yingchun.huang/.ssh/config\r\ndebug1: Applying options for *\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 4 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_request_forwards: requesting forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 8957\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 3\r\ndebug2: Remote version: 3\r\ndebug2: Server supports extension "posix-rename@openssh.com" revision 1\r\ndebug2: Server supports extension "statvfs@openssh.com" revision 2\r\ndebug2: Server supports extension "fstatvfs@openssh.com" revision 2\r\ndebug3: Sent message fd 5 T:16 I:1\r\ndebug3: SSH_FXP_REALPATH . -> /root\r\ndebug3: Looking up /home/yingchun.huang/.ansible/tmp/ansible-local-89634L0CC2/tmpkgzcn9\r\ndebug3: Sent message fd 5 T:17 I:2\r\ndebug3: Received stat reply T:101 I:2\r\ndebug1: Couldn\'t stat remote file: No such file or directory\r\ndebug3: Sent message SSH2_FXP_OPEN I:3 P:/root/.ansible/tmp/ansible-tmp-1533799417.82-278701308886841/setup.py\r\ndebug3: Sent message SSH2_FXP_WRITE I:4 O:0 S:32768\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 4 32768 bytes at 0\r\ndebug3: Sent message SSH2_FXP_WRITE I:5 O:32768 S:32768\r\ndebug3: Sent message SSH2_FXP_WRITE I:6 O:65536 S:32768\r\ndebug3: Sent message SSH2_FXP_WRITE I:7 O:98304 S:32768\r\ndebug3: Sent message SSH2_FXP_WRITE I:8 O:131072 S:32768\r\ndebug3: Sent message SSH2_FXP_WRITE I:9 O:163840 S:32768\r\ndebug3: Sent message SSH2_FXP_WRITE I:10 O:196608 S:7745\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 5 32768 bytes at 32768\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 6 32768 bytes at 65536\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 7 32768 bytes at 98304\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 8 32768 bytes at 131072\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 9 32768 bytes at 163840\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 10 7745 bytes at 196608\r\ndebug3: Sent message SSH2_FXP_CLOSE I:4\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug1: mux_client_request_session: master session id: 3\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<cm075> ESTABLISH SSH CONNECTION FOR USER: root
<cm075> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/yingchun.huang/.ansible/cp/2091d1b79e cm075 '/bin/sh -c '"'"'chmod u+x /root/.ansible/tmp/ansible-tmp-1533799417.82-278701308886841/ /root/.ansible/tmp/ansible-tmp-1533799417.82-278701308886841/setup.py && sleep 0'"'"''
<cm075> (0, '', 'OpenSSH_5.3p1, OpenSSL 1.0.1e-fips 11 Feb 2013\ndebug1: Reading configuration data /home/yingchun.huang/.ssh/config\r\ndebug1: Applying options for *\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 4 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_request_forwards: requesting forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 8957\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 3\r\ndebug1: mux_client_request_session: master session id: 3\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<cm075> ESTABLISH SSH CONNECTION FOR USER: root
<cm075> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/yingchun.huang/.ansible/cp/2091d1b79e -tt cm075 '/bin/sh -c '"'"'/usr/bin/python /root/.ansible/tmp/ansible-tmp-1533799417.82-278701308886841/setup.py && sleep 0'"'"''
```

### LINKED ISSUE #44819: Please add docs for with_first_found.
##### SUMMARY

There is no description of with_first_found in the docs. It's mentioned several times
but without any explanations.

##### ISSUE TYPE
 - Documentation Report

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

fixes #44819
##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

first_found
##### ANSIBLE VERSION

```
2.x
```

### CODE REVIEW

**Comment by mattclay** [comment]
The sanity test failure is valid. The Azure integration test failures are unrelated.

File: lib/ansible/plugins/lookup/first_found.py [hunk 1]
```diff
@@ -15,17 +15,29 @@
         to the containing role/play/include/etc's location.
       - The list of files has precedence over the paths searched.
         i.e, A task in a  role has a 'file1' in the play's relative path, this will be used, 'file2' in role's relative path will not.
+      - Either a list of files (``_terms``) or a key `files` with a list of files is required for this plugin to operate.
```
Comment (gundalow) [inline lib/ansible/plugins/lookup/first_found.py]: Assuming this is the same as modules, then I think you need `C(_terms)`, same comment applies else where

File: lib/ansible/plugins/lookup/first_found.py [hunk 2]
```diff
@@ -106,6 +118,10 @@ def run(self, terms, variables, **kwargs):
         if anydict:
             for term in terms:
                 if isinstance(term, dict):
+
+                    if 'skip' in term:
+                        display.deprecated('Use errors="ignore" instead of skip', version='2.12')
```
Comment (acozine) [inline lib/ansible/plugins/lookup/first_found.py]: Shippable is failing with "undefined variable `display`" on this line.

**Comment by samccann** [comment]
rebuild_merge

**Pull request merged**